classes/blocks/class-base-block.php

```diff
@@ -94,4 +96,93 @@ protected function is_rest_request() {
 	public static function update_data( array $fields ) {
 		throw new NotImplemented( 'Method update_data is not implemented for ' . static::class );
 	}
+
+	/**
+	 * Register scripts and styles for a block
+	 */
+	public static function enqueue_editor_assets() {
+		$camelized_block_name = static::get_camelized_block_name();
+		$url_path             = trailingslashit( P4GBKS_PLUGIN_URL ) . 'assets/build/' . $camelized_block_name;
+		$dir_path             = trailingslashit( P4GBKS_PLUGIN_DIR ) . 'assets/build/' . $camelized_block_name;
+		$block_name           = static::get_full_block_name();
+
+		// Editor Script.
+		wp_enqueue_script(
+			$block_name . '-editor-script',
+			$url_path . 'EditorScript.js',
```

Comment:
It seems simpler to give each block its own directory with all files. Then we can use generic names (`edit.js` like WP does) for the files and don't need to repeat the block name in all of them.

Reply:
I just checked their convention and it goes as follows:

`index.js` -> the place where `registerBlockType` happens.
`edit.js` -> the `edit()` function
`save.js` -> the `save()` function

Of course there is no frontend script.
Is that your proposal?

Reply:
Makes sense :+1:

Reply:
Ok, I'll go ahead with the change so we can move forward, personally, I'm not sure if I like WP's conventions, I'm not a big fan of having a bunch of files with the same name but in different folders (`index.js` was a long standing discussion in the NodeJS community), aaaand... we are doing things a bit differently than WP, for example, there is no frontend script except for the `save()` method used to generate the static (pre-hydration) markup. We'll need a different script to make the hydration work. We'll see about that in the Carousel Header PR.

Reply:
After going back to this, there is an issue: the `edit.js`, `index.js` and `save.js` convention is for the source files, not the build files. Registering blocks actually changed a lot since we implemented our blocks, I'm not sure about this change, feels like an unnecessary step at this point, and the current version is consistent with the entry points we need to build to make things run in the browser, I actually kinda like it a bit more than the WP way. 

In fact, our implementation is a bit outdated if you compare it to the current WP implementation:
https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src

So maybe we should consider updating our implementation using JSON metadata, CSS modules, etc., but I think it's something for a subsequent stage. 

I don't know what problem are we solving here, I mean, we can exchange our opinionated views about file and function names endlessly but I don't think is truly relevant.

Reply:
If you think it's better to address this later that's fine for me. It seemed to make sense at this moment since it would make the code to handle these easier to work with (less string concatenation). But seems easy to address later :ok_hand: